src/Compilers/CSharp/Portable/Symbols/Extensions/SourceExtensionImplementationMethodSymbol.cs

```diff
@@ -175,6 +206,27 @@ public override int Ordinal
                     return ordinal + 1;
                 }
             }
+
+            internal override void AddSynthesizedAttributes(PEModuleBuilder moduleBuilder, ref ArrayBuilder<CSharpAttributeData> attributes)
+            {
+                // Copy AllowNull/DisallowNull from the property onto the implementation setter's `value` parameter
+                if (this._underlyingParameter is SynthesizedAccessorValueParameterSymbol
+                    && this._underlyingParameter.ContainingSymbol is SourcePropertyAccessorSymbol { MethodKind: MethodKind.PropertySet, AssociatedSymbol: SourcePropertySymbolBase extensionProperty })
+                {
+                    foreach (CSharpAttributeData attr in extensionProperty.GetAttributes())
+                    {
+                        if (attr.IsTargetAttribute(AttributeDescription.AllowNullAttribute)
+                            || attr.IsTargetAttribute(AttributeDescription.DisallowNullAttribute))
```

Comment:
Do we have tests that explicitly verify presence/lack of these attributes in metadata?

Reply:
See `Nullability_PropertyAccess_18` `Nullability_PropertyAccess_19`. I'll add explicit check for lack in those tests too